File: fixer.py
import os
import re

def main():
    lib_dir = 'lib'
    for root, dirs, files in os.walk(lib_dir):
        for file in files:
            if not file.endswith('.dart'):
                continue
            path = os.path.join(root, file)
            with open(path, 'r', encoding='utf-8') as f:
                content = f.read()

            original = content
            
            # Fix const Theme.of
            content = re.sub(r'const\s+([a-zA-Z0-9_]+\([^)]*Theme\.of)', r'\1', content)
            content = re.sub(r'const\s+(Theme\.of)', r'\1', content)
            
            # AppColors issue
            content = content.replace('AppColors.Theme.of(context).colorScheme.surface', 'AppColors.darkPrimaryLight')
            content = content.replace('AppColors.Theme.of(context).primaryColor', 'Color(0xFF0A2314)')

            # Fix CardTheme
            content = content.replace('cardTheme: CardTheme(', 'cardTheme: CardThemeData(')

            if content != original:
                with open(path, 'w', encoding='utf-8') as f:
                    f.write(content)
                print(f"Fixed {path}")

if __name__ == '__main__':
    main()

File: fixer.dart
import 'dart:io';

void main() {
  final dir = Directory('lib');
  
  void processFile(File file) {
    if (file.path.contains('main_screen.dart')) return;
    
    var content = file.readAsStringSync();
    var original = content;

    content = content.replaceAll(RegExp(r'const\s+([a-zA-Z0-9_]+\([^)]*Theme\.of)'), r'\1');
    content = content.replaceAll(RegExp(r'const\s+(Theme\.of)'), r'\1');
    
    content = content.replaceAll('AppColors.Theme.of(context).colorScheme.surface', 'AppColors.darkPrimaryLight');
    content = content.replaceAll('AppColors.Theme.of(context).primaryColor', 'Color(0xFF0A2314)');
    content = content.replaceAll('cardTheme: CardTheme(', 'cardTheme: CardThemeData(');
    content = content.replaceAll('Theme.of(context).primaryColor', 'Color(0xFF0A2314)'); // Reverting
    content = content.replaceAll('Theme.of(context).colorScheme.surface', 'Color(0xFF1E4226)');
    content = content.replaceAll('Theme.of(context).colorScheme.primary', 'Color(0xFFF7DE9B)');
    content = content.replaceAll('Theme.of(context).colorScheme.outline', 'Color(0xFFA0B3A6)');
    content = content.replaceAll('Theme.of(context).colorScheme.secondary', 'Color(0xFFD4A373)');

    if (content != original) {
      file.writeAsStringSync(content);
      print('Fixed const in \${file.path}');
    }
  }

  dir.listSync(recursive: true).whereType<File>().where((f) => f.path.endsWith('.dart')).forEach(processFile);
}
